File: Delegate iOS ObjC/Delegate iOS ObjC/ViewController.m
#import "ViewController.h"
#import "GreenViewController.h"

@implementation ViewController

- (void) send : (NSString *)message
{
    receiveMessageLabel.text = message;
}

-(IBAction)gotoGreenAction:(id)sender
{
    [self performSegueWithIdentifier:@"greenSegueId" sender:nil];
}

-(void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender
{
    GreenViewController *greenInstance = [segue destinationViewController];
    greenInstance.delegate = self;
    
}

@end

File: delegation ios/delegation ios/ViewController.swift
import UIKit

class ViewController: UIViewController, sendingMessageDelegate
{
    func send(message:String)
    {
        receiveMessageLabel.text = message
        
    }
    
    @IBOutlet weak var receiveMessageLabel: UILabel!
    
    @IBAction func GoGreenAction(_ sender: Any)
    {
        self.performSegue(withIdentifier: "GreenSegueId", sender: nil)
    }
    
    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
        let greenInstance = segue.destination as! GreenViewController
        
        greenInstance.delegate = self
    }
}

